app/controllers/orders_controller.rb

```diff
@@ -326,7 +314,10 @@ def index
   # Group into chunks by the group id. If group_id is nil, then it should get its
   # own chunk.
   def grouped_order_details
-    @order.order_details.order(:group_id).slice_when do |before, after|
+    # Do not do any additional scopes on `@order.order_details` to ensure the list
+    # of OrderDetail objects is memoized. Validations are run on `@order.order_details`
+    # and we want to be able to render validation errors inline.
+    @order.order_details.sort_by(&:group_id).slice_when do |before, after|
       before.group_id.nil? || before.group_id != after.group_id
```

Comment:
I wonder as I look at this again if the comment indicates that the code wants to be factored a little differently, maybe pulling `@order.order_details` into a separate load method so as to force a memoized version?